Aardvark/ARKEmailBugReporter.m

```diff
@@ -342,155 +336,128 @@ - (NSString *)formattedLogMessagesAttachmentExtension;
 
 #pragma mark - Private Methods
 
-- (void)_stealFirstResponder;
-{
-    ARKInvisibleView *invisibleView = [ARKInvisibleView new];
-    invisibleView.layer.opacity = 0.0;
-    [[UIApplication sharedApplication].keyWindow addSubview:invisibleView];
-    [invisibleView becomeFirstResponder];
-    [invisibleView removeFromSuperview];
-}
-
-- (void)_showBugTitleCaptureAlert;
+- (void)_showBugReportPrompt;
 {
-    /*
-     iOS 8 often fails to transfer the keyboard from a focused text field to a UIAlertView's text field.
-     Transfer first responder to an invisble view when a debug screenshot is captured to make bug filing itself bug-free.
-     */
-    [self _stealFirstResponder];
-    
-    NSString * const title = NSLocalizedString(@"What Went Wrong?", @"Title text for alert asking user to describe a bug they just encountered");
-    NSString * const message = NSLocalizedString(@"Please briefly summarize the issue you just encountered. You’ll be asked for more details later.", @"Subtitle text for alert asking user to describe a bug they just encountered");
-    NSString * const composeReportButtonTitle = NSLocalizedString(@"Compose Report", @"Button title to compose bug report");
-    NSString * const cancelButtonTitle = NSLocalizedString(@"Cancel", @"Button title to not compose a bug report");
-    
-    UIAlertController *const alertController = [UIAlertController alertControllerWithTitle:title message:message preferredStyle:UIAlertControllerStyleAlert];
-    
-    [alertController addAction:[UIAlertAction actionWithTitle:composeReportButtonTitle style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
-        UITextField *textfield = [alertController.textFields firstObject];
-        [self _createBugReportWithTitle:textfield.text];
-    }]];
-    
-    [alertController addAction:[UIAlertAction actionWithTitle:cancelButtonTitle style:UIAlertActionStyleDefault handler:NULL]];
-    
-    [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField) {
-        [self _configureAlertTextfield:textField];
+    id <ARKEmailBugReporterPromptingDelegate> const promptPresenter = (self.promptingDelegate ?: [ARKDefaultPromptPresenter new]);
+    [promptPresenter showBugReportingPromptForConfiguration:[self _configurationWithCurrentSettings] completion:^(ARKEmailBugReportConfiguration * _Nullable configuration) {
+        if (configuration != nil) {
+            [self _createBugReportWithConfiguration:configuration];
+        }
```

Comment:
add a comment here saying that if configuration is nil we shouldn't show anything because the callee has signaled to us that we no longer need to create a bug?

It'd also be _great_ to add a test to ensure this behavior doesn't break.

Reply:
Added a comment to an `else`.  Not sure the best way to add a test for this, but open to suggestions.

Reply:
ah, darnit. Yeah our UI + bug generating code isn't really set up for mocking is it? Would be great to be able to mock the email creation... but that's a larger project (though tbh it's not that big a change).